code/src/main/android/orxAndroidSupport.cpp

```diff
@@ -221,11 +259,63 @@ extern "C" const char *orxAndroid_GetInternalStoragePath()
   return sstAndroid.app->activity->internalDataPath;
 }
 
+extern "C" void orxAndroid_JNI_GetArguments()
+{
+  JNIEnv *env = orxAndroid_JNI_GetEnv();
+
+  env->PushLocalFrame(16);
+
+  /* Gets meta data  */
+  jobject metaData = orxAndroid_JNI_getActivityMetaData(env);
+  if (metaData == NULL)
+  {
+    /* Stores the arguments */
+    *sstAndroid.zArguments = orxNULL;
+  }
+  else
+  {
+    const char *zLibName;
+    const char *zArguments;
+
+    /* Finds classes */
+    jclass bundleClass = env->FindClass("android/os/Bundle");
+
+    /* Finds methods */
+    jmethodID getStringMethod = env->GetMethodID(bundleClass, "getString", "(Ljava/lang/String;)Ljava/lang/String;");
+
+    /* Reads meta data values */
+    jstring libName = (jstring)env->CallObjectMethod(metaData, getStringMethod, env->NewStringUTF("android.app.lib_name"));
```

Comment:
I wouldn't rely on "android.app.lib_name" for the first argument

1) unlikely, one could want to use a different name for the lib and the root Config File
2) technically, this expect that orx is running from a GameActivity, if tomorrow, we want to use something else to run orx, we would need to changes this

I would only use "org.orx.lib.arguments" for all orx args, maybe then, the name should be CmdLine

Reply:
I see your point, but there's more to it.

`"android.app.lib_name"` originates from `NativeActivity`. The same logic is used there. Thus, a well-established API. I considered reading this value from `GameActivity` field, but then realized that it was copied from `NativeActivity`. Hence the constant.

The first argument to orx must be the exe name (or lib name in our case). I store it separately, allowing for empty string without handling quotes and stuff. If you put the lib name in the arguments you will run into a parse problem.

Reply:
As you said, a well-established API ... of NativeActivity, not ORX API
so we kind of bind this specific ORX API (to pass arguments) with a implementation details API, I don't think it's a good design choice.

I don't understand the parsing issue, just takes it at string, and chop it around white spaces

Reply:
I think the latest push gives us the best of two worlds.

1. If `arguments` is set, tokenize and use for `argv`
2. Else if `lib_name` is set, use it for `argv`
3. Else use empty string for `argv`, defaulting to `orx.ini`

If the user has a working Windows setup for `game.exe` and `game.ini` then getting the android app to work involves less steps. Simply name the module `"game"` and declare it as `lib_name` (as documented for `NativeActivity`) and it runs. Without (2) above, the app would not run. Been there, done that.